File: ror/public/javascripts/models/systemimagetype.js
// Ember model for systemimagetype object
var Systemimagetype = DS.Model.extend({
  name: DS.attr('string'),
  systemimages: DS.hasMany('systemimages', {async: true}),
  projects: DS.hasMany('projects', {async: true}),
});

module.exports = Systemimagetype;

File: ror/app/models/systemimagetype.rb
# Store all exploitation system whi are the same type: unix / linux / windows mainly
#
# @author Eric Fehr (ricofehr@nextdeploy.io, github: ricofehr)
class Systemimagetype < ActiveRecord::Base
  # One exploitation system is caracterised by one unique property type
  has_many :systemimages, dependent: :destroy

  # some properties are mandatory and must be well-formed
  validates :name, presence: true
end
